app/jobs/aspace_index_job.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+# Asynchronous job used to index EAD Documents
+class AspaceIndexJob < ApplicationJob
+  # Class for capturing the output of the Traject indexer
+  class EADArray < Array
+    # Appends the output of the Traject indexer
+    # (Used by Traject#foo)
+    # @param context [Traject::Context]
+    def put(context)
+      push(context.output_hash)
+    end
+  end
+
+  # Retrieve the file path for the ArcLight core Traject configuration
+  # @return [String]
+  def arclight_config_path
+    pathname = Rails.root.join("lib", "pulfalight", "traject", "ead2_config.rb")
+    pathname.to_s
+  end
+
+  # Construct a Traject indexer object for building Solr Documents from EADs
+  # @return [Traject::Indexer::NokogiriIndexer]
+  def indexer
+    indexer = Traject::Indexer::NokogiriIndexer.new(repository: @repository_id)
+    indexer.tap do |i|
+      i.load_config_file(arclight_config_path)
+    end
+  end
+
+  # Retrieve the connection to the Solr index for Blacklight
+  # @return [RSolr]
+  def blacklight_connection
+    repository = Blacklight.default_index
+    repository.connection
+  end
+
+  def logger
+    Rails.logger || Logger.new(STDOUT)
+  end
+
+  def aspace_client
+    @aspace_client ||= Aspace::Client.new
```

Comment:
This is not a criticism, but I would be curious as to whether or not a future feature might be to provide a `Configuration` Object for the `Aspace::Client` constructor. This is not needed here.

Reply:
I am sorry, please disregard this, I will delete these comments looking below.